### ISSUE TITLE
How does RecruitEvaluator work?

### ISSUE THREAD

**Issue by dozens** [lstern/SWProxy-plugins#11]
I do not see any print, file or write statements in the .py file. Nor any references to the swconfig file.

**Comment by fperegrinvs** [comment]
Yeah, still undocumented. This plugin extends the visitor plugin to include
extra information to help evaluating new recruits. Its adds 2 columns for
each rune "Current Potential" and "max potential". Current potential is a
variation of barion's formula to evaluate non +12 runes, Max potential
simulates what would be the score if rune gets perfect upgrades.

Then, it adds a summary including :

# of 6\* mobs

# of 6\* runes

# of level 15 runes

# average rune score

**Comment by dozens** [comment]
Ok. I wasnt expecting any dependencies. I didnt have the plugins from SWProxy. What other dependencies are there? I just got an exception visiting someone.

2016-03-28 14:35:50,832: root - ERROR - Exception while executing plugin "Generate Visit Friend": global name 'DictUnicodeWriter' is not defined
Traceback (most recent call last):
  File "/root/GH/SWParser/SWPlugin.py", line 42, in call_plugins
    getattr(plugin.plugin_object, func_name)(*args)
  File "/root/GH/SWParser/plugins/GenerateVisitFriend.py", line 72, in process_request
    parse_visit_data(resp_json)
  File "/root/GH/SWParser/plugins/GenerateVisitFriend.py", line 41, in parse_visit_data
    visit_writer = DictUnicodeWriter(visit_file, fieldnames=visit_fieldnames)
NameError: global name 'DictUnicodeWriter' is not defined

**Comment by dozens** [comment]
Looks like there are multiple dependencies on SWProxy. I use a forked version of parser.py and just updated my SWProxy.py to take advantage of plugins. Perhaps better decoupling between plugins here and the code base in SWProxy is needed.

**Comment by fperegrinvs** [comment]
SW Proxy had a major internal rewrite to better support plugins. Csv files
generation now uses DicUnicodeWritter to make code cleaner, more robust and
to allow plugins to extend the CSVs. I understand you concern about new SW
Proxy versions breaking existing plugins and while a total code decoupling
is not desired (SW Proxy have useful methods for rune and mob parsing) we
will try to minimize internal changes that could break existing plugins.
Its not fair to mix pre-plugin and post-plugin code and complain about
issues but I got the message.

**Comment by dozens** [comment]
I dont think I was clear. I updated my SWProxy fix the for the new SW release, I left my parser.py intact. I saw that SWProxy had code for plugin support so I started playing around with the run logger, etc. It wasnt until I used RecruitEvaluator I had to make major changes to my parser.py. I wasnt using any plugins previously.

If a plugin has a dependency on SWProxy shouldnt it be included in that code base?
